crates/subspace-service/src/dsn/node_provider_storage.rs

```diff
@@ -0,0 +1,62 @@
+use std::borrow::Cow;
+use subspace_networking::libp2p::kad::record::Key;
+use subspace_networking::libp2p::kad::ProviderRecord;
+use subspace_networking::libp2p::PeerId;
+use subspace_networking::ProviderStorage;
+
+pub(crate) struct NodeProviderStorage<FixedProviderStorage, PersistentProviderStorage> {
+    // provider records from local cache
+    fixed_provider_storage: FixedProviderStorage,
```

Comment:
`fixed_provider_storage` still doesn't make any sense to be as a name. Also please use `///` instead of `//` as we discussed before.

What about  `implicit_provider_storage` and `persistent_provider_storage`?